platform/src/Playground.js

```diff
@@ -867,6 +900,39 @@ function savePanelContents(event){
     });
 }
 
+/**
+ * Poll for editor to become available. 
+ * @param {*} statusUrl
+ * @param {*} url 
+ * @param {*} editorPanel 
+ * @param {*} editorActivity TODO remove as this can be found to save having to specify in config.
+ */
+async function checkEditorReady(statusUrl, url, editorPanel, editorActivity){
+
+   let response  = await fetch(statusUrl);
+
+   if (response.status == 200){ 
+        const result = await response.json();
+
+        if (!result.editorReady){
+            await new Promise(resolve => setTimeout(resolve, 2000));
+            await checkEditorReady(statusUrl, url, editorPanel, editorActivity);
+
+        } else {
+            // Successful 
+            console.log("Editor ready.");
+            sessionStorage.setItem( editorPanel , url );
```

Comment:
That's very specific. It's OK for now, but it may make it difficult to do multiple languages or similar at some point in the future... It's definitely important to document this approach carefully that ties us to one editorPanel.

Reply:
It is the id to the panel so it is not specifically an Xtext panel but there is an assumption that we will be able to generalise the XtextEditorPanel class into something for all language workbenches. Otherwise this will have to be revised. I'll update the parameter names for now.

Reply:
Names updated in 16a9b13ece770c7d5f42407c71ab2bba858a2b63